### PR TITLE
Cache external links that have worked

### PR DESCRIPTION
**Pull request by rictic**
This way, our CI doesn't fail if an external site is being flaky or rate limiting us, but we still check all of our internal links, and we check every new link that we add to the site.

### CODE REVIEW

**Comment by aomarks** [comment]
One of the main ways we get 404s is links that worked when we first added them, but have since been moved or deleted. So won't this prevent us from noticing those cases?

**Comment by rictic** [comment]
Hm, we could periodically clear the list

**Comment by aomarks** [comment]
> Hm, we could periodically clear the list

As discussed offline, maybe a daily job that ignores this list. Ideally it would file a bug if it finds a problem (and update/delete it if the status changes independently of us doing anything).

But we can do that some other time. Maybe just file an issue?

This PR is good because we'll probably start paying attention to new failures, at least.

**Review by aomarks — CHANGES_REQUESTED** [review]

File: packages/lit-dev-tests/src/check-broken-links.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
+import blc from 'broken-link-checker';
```
[the remark this answers is not in the document]
Comment (rictic) [reply packages/lit-dev-tests/src/check-broken-links.ts:7]: Done

File: packages/lit-dev-tests/src/check-broken-links.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        knownGoodUrls.add(result.url.resolved);
+      },
+    }
+  );
+  checker.enqueue('http://localhost:6415', {});
+  return donePromise;
+}
+
+main().catch((e) => {
+  console.error(e);
```
[the remark this answers is not in the document]
Comment (rictic) [reply packages/lit-dev-tests/src/check-broken-links.ts]: Done

File: packages/lit-dev-tests/src/check-broken-links.ts [hunk 3]
```diff
@@ -1,3 +1,9 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
```
Comment (aomarks) [inline packages/lit-dev-tests/src/check-broken-links.ts:3]: 2022

**Review by aomarks — CHANGES_REQUESTED** [review]

File: packages/lit-dev-tests/src/check-broken-links.ts (hunk 3 above)
Comment (rictic) [reply packages/lit-dev-tests/src/check-broken-links.ts:3]: Eep, yep!

**Review by aomarks — APPROVED** [review]

**Pull request merged**